### PR TITLE
Initial Integration of skeleton-generic

### LINKED ISSUE #294: Integrate skeleton-generic at a Minimum
# 🚀 Feature Proposal

The bare minimum should be integrating [cisagov/skeleton-generic](https://github.com/cisagov/skeleton-generic) to get the updated linting configuration. Given the monolithic nature of this repository and the way it is organized, I'm not sure if one of the other skeletons would be a viable parent.

## Motivation

There is minimum automatic linting/testing done in this repository, and at the minimum we should be linting to keep everything here in line with our other active projects.

## Example

This would allow GitHub Actions usage and [action-lineage](https://github.com/cisagov/action-lineage) automation.

## Pitch

Every actively developed project should be derived from a skeleton project and have the appropriate suite of CI/CD in place.

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This PR is an initial integration of [cisagov/skeleton-generic](https://github.com/cisagov/skeleton-generic).

## 💭 Motivation and Context ##

I'm moving forward with working on #301 , and this gets the framework for everything in place to work on passing each `pre-commit` hook group and merging those in their own PRs.

## 🧪 Testing ##

The GitHub Actions workflow is passing with the pared down `pre-commit` configuration to align with the progress made in #295 .

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] _All_ future TODOs are captured in issues, which are referenced in code comments.
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] Tests have been added and/or modified to cover the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,10 @@
...
-playbook.retry
+__pycache__
+.coverage
+.mypy_cache
+.pytest_cache
 .python-version
 .terraform
-terraform.tfstate*
 *.tfvars
-*.dyn.tf
```
Comment (jsf9k) [inline .gitignore:7]: Do we still need the `*.dyn.tf` line for the code generated in the `terraform/dyn_*` directories?  I'm not sure, so I'm asking.

**Review by jsf9k — APPROVED** [review]
This looks good to me!  Thanks for dragging this 900 pound gorilla into the light!

I had one question, but I am happy to approve this no matter the answer.

File: .gitignore (hunk 1 above)
Comment (mcdonnnj) [reply .gitignore:7]: Ah, that's a good question. We probably do and I just got overzealous with `.gitignore` cleaning.

Comment (mcdonnnj) [reply .gitignore:7]: I have restored that pattern in 22c028b

**Review by hillaryj — APPROVED** [review]
💀

**Pull request merged**